grc/utils/application_progress.py

```diff
@@ -1,4 +1,5 @@
 from datetime import datetime
+import ast
```

Comment:
json is unreliable, it doesn't handle json with single/double quotes mixed. ast handles this better (https://stackoverflow.com/questions/12735106/python-ast-vs-json-for-str-to-dict-translation)

Reply:
I don't think this is relevant, since we'll never be parsing hand-written json.
The json would only ever be written and read by these `json.dumps` and `json.loads`.
But I don't mind just going with `ast` instead - that should work too.